dm/syncer/syncer.go

```diff
@@ -3781,3 +3815,58 @@ func (s *Syncer) flushOptimisticTableInfos(tctx *tcontext.Context) {
 		tctx.L().Error("failed to flush table points with table infos", log.ShortError(err))
 	}
 }
+
+func (s *Syncer) setGlobalPointByTime(tctx *tcontext.Context, timeStr string) error {
```

Comment:
we should enable safe-mode between `position of timestamp` and `previous checkpoint`, right? didn't see any logic about this